extensions/interactions/TextInput/TextInputPredictionServiceSpec.js

```diff
@@ -29,6 +29,25 @@ describe('Text Input Prediction Service', function() {
       $rootScope = $injector.get('$rootScope');
       $scope = $rootScope.$new();
     }));
+    it('should predict the same as oppia-ml', function() {
+      jasmine.getJSONFixtures().fixturesPath = 'base/core/tests/data';
+      var classifierData = getJSONFixture('text_input_classifier_data.json');
+      var trainingData = getJSONFixture('text_classifier_results.json');
+      var correctPredictions = 0, totalAnswers = 0;
+
+      for (var i = 0; i < trainingData.length; i++) {
+        for (var j = 0; j < trainingData[i].answers.length; j++) {
+          predictedAnswerGroup = predictionService.predict(
+            classifierData, trainingData[i].answers[j].answer);
+          if (predictedAnswerGroup === -1) {
+            predictedAnswerGroup = 3;
```

Comment:
I don't understand why do this? First of all, why predicted answer group is -1 and why change it to 3?

Reply:
Added a comment as well as a small change in the textpredictor frontend service to reflect this. We discussed in email how when the SVM in the backend was given an empty(zero) vector to predict upon, it predicted 3. So to maintain consistency, our frontend should also guess 3 when given an empty vector. The TextPredictor frontend service used to return -1 when given an empty vector, so I changed it to return 3 when given an empty vector.